packages/labs/signals/src/lib/watch.ts

```diff
@@ -0,0 +1,73 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {DirectiveResult, Part, directive} from 'lit/directive.js';
+import {AsyncDirective} from 'lit/async-directive.js';
+import {Signal} from 'signal-polyfill';
+import {SignalWatcher} from './signal-watcher.js';
+
+export class WatchDirective<T> extends AsyncDirective {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private __host?: SignalWatcher;
+  __signal?: Signal.State<T> | Signal.Computed<T>;
+  private __watcher = new Signal.subtle.Watcher(() => {
+    this.__host?.updateWatch(this as WatchDirective<unknown>);
+  });
+
+  commmit() {
+    this.setValue(Signal.subtle.untrack(() => this.__signal?.get()));
```

Comment:
Again here, I think you have to call `this.__watcher.watch()` to reset the watcher.

Reply:
done